lib/shared/src/mentions/providers/urlMentions.ts

```
@@ -1,49 +1,71 @@
             return []
         }
-        return [
-            {
-                type: 'file',
-                uri: url,
-                content,
-                title: tryGetHTMLDocumentTitle(content),
-                source: ContextItemSource.Uri,
-            },
-        ]
-    } catch (error) {
-        // Suppress errors because the user might be typing a URL that is not yet valid.
-        return []
-    }
-}
 
-export function isURLContextItem(item: Pick<ContextItem, 'uri'>): boolean {
-    return item.uri.scheme === 'http' || item.uri.scheme === 'https'
+        try {
+            const content = await fetchContentForURLContextItem(url.toString(), false, signal)
+            if (content === null) {
+                return []
+            }
+            return [
+                {
+                    type: 'file',
+                    uri: url,
+                    content,
+                    title: tryGetHTMLDocumentTitle(content),
+                    source: ContextItemSource.Uri,
+                    provider: 'url',
+                },
+            ]
+        } catch (error) {
+            // Suppress errors because the user might be typing a URL that is not yet valid.
+            return []
+        }
+    },
+
+    async resolveContextItem(item, signal) {
+        if (item.content !== undefined) {
+            return [item as ContextItemWithContent]
+        }
+        const content = await fetchContentForURLContextItem(item.uri.toString(), true, signal)
+        return [{ ...item, content }]
+    },
 }
 
-export async function fetchContentForURLContextItem(
+async function fetchContentForURLContextItem(
+    url: string,
+    throwIfNotOK: true,
+    signal?: AbortSignal
+): Promise<string>
+async function fetchContentForURLContextItem(
+    url: string,
+    throwIfNotOK: false,
+    signal?: AbortSignal
+): Promise<string | null>
+async function fetchContentForURLContextItem(
     url: string,
+    throwIfNotOK: boolean,
```

it seems from the callsites you don't need this. The only place you set this to false you catch all exceptions anyway and return the same thing.